src/client/datascience/interactive-window/nativeInteractiveWindowProvider.ts

```diff
@@ -238,19 +236,8 @@ export class NativeInteractiveWindowProvider implements IInteractiveWindowProvid
     }
 
     private async getControllerForInteractiveWindow(): Promise<string | undefined> {
-        // Fetch the active interpreter and use the matching controller
-        const api = await this.pythonApi.getApi();
```

Comment:
The logic for what's the preferred should be in the manager (that's where it is today for native notebooks).